### PR TITLE
Returned User TermsOfService Data in Certain GetUser APIs To Be Used on Webapp for Decicion Making

### PR DESCRIPTION
**Pull request by harshilsharma**
#### Summary
Returned User TermsOfService Data in Certain GetUser APIs To Be Used on Webapp for Decicion Making

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] Added API documentation (required for all new APIs)
- [ ] All new/modified APIs include changes to the drivers
- [x] Has UI changes - Redux - https://github.com/mattermost/mattermost-redux/pull/797
- [ ] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/mattermost-server/blob/master/i18n/en.json)) updates

WIP: pending test cases and further testing

### CODE REVIEW

File: api4/user.go [hunk 1]
```diff
@@ -119,6 +119,17 @@ func getUser(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	userTermsOfService, err := c.App.GetUserTermsOfService(user.Id)
```
Comment (lieut-data) [inline api4/user.go]: Can you change this to only fetch and return the terms of service if `c.IsSystemAdmin()` or `c.App.Session.UserId == user.Id`? We shouldn't return this information about other users unless requested by a system admin.

Comment (harshilsharma) [reply api4/user.go]: Done.

**Comment by harshilsharma** [comment]
@lieut-data I've made the requested changes.

**Comment by amyblais** [comment]
A quick note that tomorrow (v5.10 Feature Complete) is the last day to merge feature PRs if we want this for v5.10.

File: api4/user.go [hunk 2]
```diff
@@ -149,6 +162,17 @@ func getUserByUsername(c *Context, w http.ResponseWriter, r *http.Request) {
...
```
Comment (grundleborg) [inline api4/user.go]: You need the `if c.IsSystemAdmin() || c.App.Session.UserId == user.Id {` check here too I think?

Comment (harshilsharma) [reply api4/user.go]: Done.

**Comment by amyblais** [comment]
@harshilsharma Would you be open to helping resolving the conflicts here and then this can be merged and cherry picked to v5.10?

**Comment by hanzei** [comment]
@amyblais Actually there is not conflict here. This can be merged right away.

**Comment by jasonblais** [comment]
API docs here https://github.com/mattermost/mattermost-api-reference/pull/432

**Pull request merged**